Zend/zend_attributes.c

```diff
@@ -70,12 +64,38 @@ ZEND_API zend_attribute *zend_get_attribute_str(HashTable *attributes, const cha
 	return NULL;
 }
 
+static void attribute_ptr_dtor(zval *v) /* {{{ */
+{
+	zend_attribute_free((zend_attribute *) Z_PTR_P(v));
+}
+/* }}} */
+
 ZEND_API void zend_compiler_attribute_register(zend_class_entry *ce, zend_attributes_internal_validator validator)
 {
+	if (ce->type != ZEND_INTERNAL_CLASS) {
+		return;
```

Comment:
This should throw an error using `zend_error_noreturn()` instead. This case will likely never happen, but it is a fatal error if it does.